portal-impl/src/com/liferay/portlet/documentlibrary/service/permission/DLFileEntryPermission.java

```diff
@@ -110,6 +118,9 @@ public static boolean contains(
 			if (hasPermission != null) {
 				return hasPermission.booleanValue();
 			}
+			else if (actionId.equals(ActionKeys.VIEW)) {
```

Comment:
Thanks @Alec-Shay, I've sent you a PR with changes that make the reasons for the fix a bit more explicit.

Thanks!